File: PadelClub.Services/Database/Tournament.cs
using System;

namespace PadelClub.Services.Database
{
    public class Tournament
    {
        public int Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public DateTime RegistrationDeadline { get; set; }
        public int MaxParticipants { get; set; }
        public decimal EntryFee { get; set; }
        public string Status { get; set; } = "Upcoming"; // Upcoming, Ongoing, Completed, Cancelled
        public string? PrizeInfo { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime? UpdatedAt { get; set; }

        // Navigation properties
        public virtual ICollection<Match> Matches { get; set; } = new List<Match>();
        public virtual ICollection<TournamentParticipant> Participants { get; set; } = new List<TournamentParticipant>();
    }
}

File: UI/padelclub_desktop/lib/features/tournament/domain/entities/tournament.dart
class Tournament {
  const Tournament({
    this.id,
    required this.name,
    required this.description,
    required this.startDate,
    required this.endDate,
    required this.registrationDeadline,
    required this.maxParticipants,
    required this.entryFee,
    required this.status,
    this.prizeInfo,
  });

  final int? id;
  final String name;
  final String description;
  final DateTime startDate;
  final DateTime endDate;
  final DateTime registrationDeadline;
  final int maxParticipants;
  final double entryFee;
  final String status;
  final String? prizeInfo;
}
